src/gluonts/torch/model/lag_tst/module.py

```diff
@@ -0,0 +1,342 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import Tuple, Optional, List
+
+import numpy as np
+from pandas.tseries.frequencies import to_offset
+import torch
+from torch import nn
+
+from gluonts.core.component import validated
+from gluonts.model import Input, InputSpec
+from gluonts.torch.distributions import StudentTOutput
+from gluonts.torch.scaler import StdScaler, MeanScaler, NOPScaler
+from gluonts.torch.util import unsqueeze_expand, lagged_sequence_values
+from gluonts.time_feature import norm_freq_str
+from gluonts.time_feature.lag import _make_lags
+
+
+def get_lags_for_frequency(
```

Comment:
Should we use https://github.com/awslabs/gluonts/blob/dev/src/gluonts/time_feature/lag.py#L29 instead?

Reply:
we can but i wanted to avoid the default `[1, 2, 3, 4,..]` lags that are in there

Reply:
We could add an argument to control this, something like
```python
def get_lags_for_frequency(
    freq_str: str, 
    lag_ub: int = 1200, 
    num_default_lags: int = 7,  # number of lags that are always included
    num_lags: Optional[int] = None,
) -> List[int]:
    ...
    # flatten lags list and filter
    lags = [
        int(lag) for sub_list in lags for lag in sub_list if num_default_lags < lag <= lag_ub
    ]
    lags = list(range(1, num_default_lags + 1)) + sorted(list(set(lags)))
    return lags
```

Reply:
hehe that is exactly what i did just now haha but i had:

```py
def get_lags_for_frequency(
    freq_str: str,
    lag_ub: int = 1200,
    num_lags: Optional[int] = None,
    default_lags: List[int] = [1, 2, 3, 4, 5, 6, 7],
) -> List[int]:
```

Reply:
I would prefer a tuple or an int here as the default to avoid potential problems with [mutable default arguments](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments). This won't be a problem currently, but could introduce a bug if some change is made to the function in the future.

Reply:
sure sure! changing!